core/queryparser/sparql/src/main/java/org/eclipse/rdf4j/query/parser/sparql/ast/SyntaxTreeBuilder.java

```diff
@@ -11016,6 +11016,11 @@ private Token jj_consume_token(int kind) throws ParseException {
 	}
 
 	static private final class LookaheadSuccess extends java.lang.Error {
+
+		@Override
+		public synchronized Throwable fillInStackTrace() {
+			return null;
+		}
```

Comment:
Not sure if this is a generated file. This is essentially a small performance improvement for code that generates a lot of SPARQL queries.

Reply:
It is a generated file, as indicated by the comment at the top, so this will be overwritten whenever we next generate the parser from the grammar. 

I'm also not sure what the consequences of this change are. What happens on errors in sparql parsing? Are we suddenly leaving out neccessary information from some SPARQL parser errors? Why  is this even a performance considerination (I am assuming that this only happens when there is a parse error?)?

Reply:
It's the parser abusing exceptions for control flow. Not a parser error.